crowbar_framework/app/models/node_object.rb

```diff
@@ -1585,16 +1585,25 @@ def unique_device_for(device)
       end
       result = disk_lookups.map do |type|
         if meta["disks"][type] and not meta["disks"][type].empty?
-          "#{type}/#{meta["disks"][type].first}"
+          meta["disks"][type]
         end
       end
-
+      unless result.empty?
+        result = result[0]
+        match = result.find { |b| b =~ /^wwn-/ } ||
+          result.find { |b| b =~ /^scsi-[a-zA-Z]/ } ||
+          result.find { |b| b =~ /^scsi-[^1]/ } ||
+          result.find { |b| b =~ /^scsi-/ } ||
+          result.find { |b| b =~ /^ata-/ } ||
+          result.find { |b| b =~ /^cciss-/ } ||
+          result.first
```

Comment:
No, the problem is the meta["disk"][type] returns a nested list with all the elements in the [0] position.